### DISCUSSION TITLE
French translation

### DISCUSSION THREAD

**Discussion by devkatalist** [SethClydesdale/genki-study-resources#160] [Ideas]
Hello. You have done some excellent work and I am very grateful to you. I'd like to complement your efforts with a tiny contribution of mine and that is translating this project into french. What do you think?

**Comment by SethClydesdale** [comment]
Hello @lambdacore12,

There are quite a lot of contents to translate, so if you are up to the task you have my permission to create a French version of this repo. It will require a lot of work, however, so you of course do not have to feel obligated to do so. I appreciate the sentiment alone, so thank you.

Currently there is no translation framework setup, which would require editing multiple HTML files and possibly some CSS and JavaScript as well. The finished result could perhaps be merged under a /lang/fr/ directory.

If you've any questions feel free to let me know.

**Comment by devkatalist** [comment]
Thank you for your quick reply @SethClydesdale,

It's funny that you said that it requires a lot of work seeing that you sunk in quite the effort yourself! I really appreciate your efforts and I would like to contribute in helping you make a change. Of course, I know that it will not be an easy task, but I will follow on your example and do the best I can to the extent of my power.

I have one question though: where do I start?

**Comment by SethClydesdale** [comment]
Haha, that is true. It took about 3.5 years to complete both the 2nd and 3rd edition resources.

You would have to fork this repo, then clone it locally to your system. You can also edit it online too, though editing locally will allow you to preview and test your changes in browser as you make them. Once cloned there's a number of things you can do such as editing:
- HTML files
- CSS files that may contain text strings
- JS files that may contain text strings (these are more common than CSS)
- MD (markdown) files, such as the README
- any other files with texts that may be present which I'm forgetting

You can edit the majority of these files in any IDE, such as Visual Studio, Brackets, Notepad++, etc. For CSS and JS, you will have to minify the code and update the minified files for changes to take effect. I normally use Brackets with [this extension](https://github.com/abagshaw/brackets-minifier), but I'm sure there's one available for your preferred IDE as well.

For bulk editing multiple files at once (such as lesson HTML files), I used this [handy tool](https://github.com/zzzprojects/findandreplace).

For cloning and pushing changes online, you can use [GitHub desktop](https://desktop.github.com/).

**Comment by devkatalist** [comment]
Wow! That's a lot of work! Not that I was not expecting this but the fact is I have to tell you that I have never contributed to a repository on GitHub before, so I need some time to learn about it then I'll start on the project. Sorry about that...

**Comment by SethClydesdale** [comment]
It is no problem! Take your time to get acquainted with how everything works on GitHub. You can find a number of guides [here](https://docs.github.com/en) and [here](https://docs.github.com/en/desktop) for the desktop version.

**Comment by devkatalist** [comment]
Thanks. I'll let you know as soon as I'm ready.

**Comment by brice** [comment]
Hello,

Thanks for your work @SethClydesdale 

I'm also interested in a french translation of your tool. I check forked repositories and found this project : https://github.com/damienBAUDET/genki-study-resources-fr

It seems not active anymore so I forked it and I will try to provide main page translation here : https://github.com/brice/genki-study-resources-fr

I know this is a massive work but I hope it will help people.

**Reply by SethClydesdale** [reply]
Hello, @brice!

I appreciate you taking the time to carry on the work for this project! If you have any questions, please don't hesitate to get in contact with me. In the meantime, I'll look into adding this into the README so that more people can find it.

I'll be wishing you all the best!

**Comment by brice** [comment]
Hello,

During the vocabulary translation, I noticed that you mention two words that's not in my french edition of Genki. Instead of having "house ; home" and "wife" I have "rental" (location in french) and "chores" (tâches ménagères). 

What's it's on page 335 of the english version for the kanji 家 ?

**Reply by SethClydesdale** [reply]
Hi!

Here's what it looks like in the English version (3rd ed.)
![Screenshot 2025-05-19 120056](https://github.com/user-attachments/assets/f9bda6f7-ed99-4e90-8139-24ac2f2d7420)

I'm assuming they may have changed the vocab for the French version? If so, you can exclude (or keep) the English words while adding in the French ones if you like.

**Reply by brice** [reply]
Thanks, I choose to stick to the manual so I kept the french ones and excluded the english words. It was  the first time i saw a difference.

**Comment by Paulsan2** [comment]
Bonjour à tous, 
Merci pour ce projet, notamment la traduction française ! Je ne sais pas si il est toujours d'actualité mais je me permets de vous contacter car je dispose des manuels Genki I et II 3ème édition en français en PDF, si jamais cela peut vous intéresser pour avancer dans le projet je peux vous les envoyer !
Bonne journée et merci pour votre travail!
